source/interprocedural_analyses/taint/taintConfiguration.ml

```diff
@@ -223,9 +223,35 @@ let parse source_jsons =
     | `String "parametric" -> AnnotationParser.Parametric
     | unexpected -> failwith (Format.sprintf "Unexpected kind %s" (Json.Util.to_string unexpected))
   in
+  let unknown_keys = String.Hash_set.create () in
+  let check_keys ~required_keys ~valid_keys ~current_keys ~section =
+    let valid_keys_hash_set = String.Hash_set.of_list valid_keys in
+    let current_keys_hash_set = String.Hash_set.of_list current_keys in
+    let check_required_key_present key =
+      Hash_set.mem current_keys_hash_set key
+      |> function
+      | false ->
+          failwith (Format.sprintf "Required key %s is not found in section `%s`" key section)
```

Comment:
We should probably return a proper error instead of throwing an exception. That could be a follow-up.

Reply:
is it okay if I make this an asset bool like we do for model_verification errors?🤔